src/puppetlabs/puppetdb/queue.clj

```diff
@@ -132,26 +133,30 @@
   certname if the certname is long or contains filesystem special characters."
   ([] (metadata-serializer puppetdb-command->metadata-command))
   ([puppetdb-command->metadata-command]
-   (fn [received {:keys [producer-ts command version certname]}]
+   (fn [received {:keys [producer-ts command version certname compression]}]
      (when-not (puppetdb-command->metadata-command command)
        (throw (IllegalArgumentException. (trs "unknown command ''{0}''" command))))
      (let [certname (or certname "unknown-host")
            received+producer-time (encode-command-time received producer-ts)
            short-command (puppetdb-command->metadata-command command)
-           safe-certname (embeddable-certname received short-command version certname)]
-       (if (= safe-certname certname)
-         (format "%s_%s_%d_%s.json" received+producer-time short-command version safe-certname)
-         (let [name-hash (kitchensink/utf8-string->sha1 certname)]
-           (format "%s_%s_%d_%s_%s.json"
-                   received+producer-time short-command version safe-certname name-hash)))))))
+           safe-certname (embeddable-certname received short-command version certname)
+           name (if (= safe-certname certname)
+                  safe-certname
+                  (format "%s_%s"
+                          safe-certname
+                          (kitchensink/utf8-string->sha1 certname)))
+           extension (str "json" (if (not-empty compression)
```

Comment:
I'm in favor of ".json.gz" just because it's clear from the user perspective what's going on there. We have code to protect ourselves if the name is too long (we'll hash it), and adding 3 chars seems ok to me.